helper-classes/int-list-reader.h

```diff
@@ -0,0 +1,18 @@
+// Advent of Code 2020
+
+#ifndef HELPER_CLASSES_INT_LIST_READER_H_
+#define HELPER_CLASSES_INT_LIST_READER_H_
+
+#include "list-reader.h"
+
+namespace data {
+
+class IntListReader : public ListReader<int> {
```

Comment:
Also consider: instead of having a reader that takes a filename, use a static namespace `list_reading` and have a template function `template<typename T> ReadList(const char* filename)` with a `int` specialization.

In modern C++, functions are [first class](https://en.wikipedia.org/wiki/First-class_function) and don't need to be bundled into a class necessarily, and namespaces have replaced static classes.

Reply:
That's a good point, and reading in the data would have a lot of less overhead by not being wrapped by a function.